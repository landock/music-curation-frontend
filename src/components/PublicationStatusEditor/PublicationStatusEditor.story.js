import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { linkTo } from '@storybook/addon-links';
import PublicationStatusEditor from '.';

let storyProps = {
  created_at: '2018-04-30T22:09:50.997Z',
  updated_at: '2018-04-30T22:09:50.997Z',
  scheduled_at: '2018-04-30T22:09:50.997Z',
  environment: ['development'],
  published: false,
};

storiesOf('PublicationStatusEditor', module).add('default', () => (
  <PublicationStatusEditor {...storyProps} />
));
