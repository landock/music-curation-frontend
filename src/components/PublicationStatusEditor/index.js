import React from 'react';
import PropTypes from 'prop-types';

function PublicationStatusEditor({
  created_at,
  updated_at,
  scheduled_at,
  environment,
  published,
}) {
  return (
    <div>
      <p>Status: {published ? 'Scheduled' : 'Draft'}</p>
      <p>Updated: {updated_at}</p>
      <p>Scheduled: {created_at}</p>
      <p>Expires: {scheduled_at}</p>
      <div>
        <p>Exists in: </p>
        <p>
          {environment.map((env, index) => <span key={index}> {env} </span>)}
        </p>
      </div>
    </div>
  );
}

PublicationStatusEditor.propTypes = {
  created_at: PropTypes.string,
  updated_at: PropTypes.string,
  scheduled_at: PropTypes.string,
  environment: PropTypes.arrayOf(
    PropTypes.oneOf(['staging', 'production', 'development'])
  ),
  published: PropTypes.bool,
};

export default PublicationStatusEditor;
