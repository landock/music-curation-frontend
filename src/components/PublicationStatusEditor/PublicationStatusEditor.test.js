
import React from 'react'
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer'
import PublicationStatusEditor from '.'

it('PublicationStatusEditor: default', () => {
  const component = renderer.create(<PublicationStatusEditor />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})