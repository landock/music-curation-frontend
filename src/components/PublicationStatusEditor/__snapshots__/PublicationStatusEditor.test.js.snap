// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicationStatusEditor: default 1`] = `
<div>
  edit me: at components/PublicationStatusEditor/index.js
</div>
`;
